apps/hash-frontend/src/shared/notifications-context.tsx

```diff
@@ -403,8 +419,58 @@ export const NotificationsContextProvider: FunctionComponent<
               triggeredByComment,
               triggeredByUser,
             } satisfies NewCommentNotification;
-          }
+          } else if (
+            entityTypeId ===
+            systemEntityTypes.graphChangeNotification.entityTypeId
+          ) {
+            const occurredInEntityLink = outgoingLinks.find(
+              isLinkAndRightEntityWithLinkType(
+                systemLinkEntityTypes.occurredInEntity.linkEntityTypeId,
+              ),
+            );
+
+            if (!occurredInEntityLink) {
+              throw new Error(
+                `Graph change notification "${entityId}" is missing required links`,
+              );
+            }
```

Comment:
Mentioned in slack, but I think this is going to cause a crash at runtime when a draft entity is discarded in the drafts page, as it will also archive all incoming links which it will be able to do as we don't have link validation preventing required links from being removed if I recall correctly.

Reply:
We could either handle the link being missing gracefully here, or archive the notification when the draft is discarded so that it isn't being rendered here.

Reply:
I will do this in follow-up